tools/halide_image.h

```diff
@@ -223,9 +227,14 @@ class Image {
         contents->buf.min[2] = z;
         contents->buf.min[3] = w;
     }
+
+    void info(const char *tag = "Image") const;
+    void dump(const char *tag = "Image") const;
```

Comment:
I think it would be preferable to have these be standalone functions, rather than members of Image. So, the usage would be:

    dump(img, "Image");

instead of

    img.dump("Image");

This would also meant that the preprocessor macros would not be necessary.